plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintCheck.kt

```diff
@@ -0,0 +1,128 @@
+package org.jlleitschuh.gradle.ktlint
+
+import net.swiftzer.semver.SemVer
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.FileTree
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Classpath
+import org.gradle.api.tasks.Console
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.OutputFiles
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.SkipWhenEmpty
+import org.gradle.api.tasks.TaskAction
+import org.jlleitschuh.gradle.ktlint.reporter.KtlintReport
+import org.jlleitschuh.gradle.ktlint.reporter.ReporterType
+import java.io.File
+import javax.inject.Inject
+
+@CacheableTask
+open class KtlintCheck @Inject constructor(objectFactory: ObjectFactory) : DefaultTask() {
```

Comment:
We love annotations ;)
Is there a less verbose way to do constructor injection in Kotlin?

Reply:
Nope, this is the way to do constructor injection.

I'm actually really pleased that in Kotlin, constructor injection is actually simpler than field injection.

Because of this, it encourages developers to use Constructor injection.